// components/ColorFulSpinLoader/ColorFulSpinLoader.js
import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

const loading = keyframes`
   0% { 
   transform: rotate(0deg); 
   }
  100% { 
  transform: rotate(360deg); 
  }
`;

const ColorFulSpin = styled.div`
  border: ${props => `16px solid ${props.leftColor}`};
  border-radius: 50%;
  border-top: ${props => `16px solid ${props.topColor}`};
  border-right: ${props => `16px solid ${props.rightColor}`};
  border-bottom: ${props => `16px solid ${props.bottomColor}`};
  font-size: ${props => `${props.size}px`};
  height: 10em;
  margin: 60px auto;
  position: relative;
  text-indent: -9999em;
  transform: translateZ(0);
  width: 10em;
  &:after {
    border-radius: 50%;
    height: 10em;
    width: 10em;
  }
  animation: ${props => `${loading} ${props.duration}s infinite linear;`};
`;


const ColorFulSpinLoader = props => (
  <ColorFulSpin {...props} />
);

ColorFulSpinLoader.propTypes = {
  topColor: PropTypes.string,
  rightColor: PropTypes.string,
  bottomColor: PropTypes.string,
  leftColor: PropTypes.string,
  duration: PropTypes.number,
  size: PropTypes.number,
};

ColorFulSpinLoader.defaultProps = {
  topColor: 'blue',
  rightColor: 'red',
  bottomColor: 'green',
  leftColor: 'yellow',
  duration: 1.5,
  size: 10,
};

export default ColorFulSpinLoader;
